charts/opensearch/templates/statefulset.yaml

```diff
@@ -4,22 +4,14 @@ kind: StatefulSet
 metadata:
   name: {{ template "opensearch.uname" . }}
   labels:
-    heritage: {{ .Release.Service | quote }}
-    release: {{ .Release.Name | quote }}
-    chart: "{{ .Chart.Name }}"
-    app: "{{ template "opensearch.uname" . }}"
-    {{- range $key, $value := .Values.labels }}
-    {{ $key }}: {{ $value | quote }}
-    {{- end }}
+    {{- include "opensearch.labels" . | nindent 4 }}
   annotations:
     majorVersion: "{{ include "opensearch.majorVersion" . }}"
 spec:
   serviceName: {{ template "opensearch.uname" . }}-headless
   selector:
     matchLabels:
-      release: {{ .Release.Name | quote }}
-      chart: "{{ .Chart.Name }}"
-      app: "{{ template "opensearch.uname" . }}"
+      {{- include "opensearch.selectorLabels" . | nindent 6 }}
```

Comment:
This change breaks upgrades for statefulsets. I just ran into this upgrading from `1.0.8` to `1.5.4`

Reply:
```shell
Error: UPGRADE FAILED: cannot patch "opensearch-1-master" with kind StatefulSet: StatefulSet.apps "opensearch-1-master" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

Reply:
Yes sorry this PR should have been a major version bump.

Reply:
Agreed.

Reply:
@smlx I was right!

Reply:
yes I didn't consider upgrading from previous versions - only going forward from this change.

Reply:
I was able to delete the statefulset and rerun `helm upgrade` as the reclaim policy of my PVCs were set to `Retain` so thankfully there was no data loss... **be sure to check this first to avoid accidental deletion...** :skull_and_crossbones:

Reply:
thanks @acjohnson for sharing your workaround and sorry for the breakage! :pray: